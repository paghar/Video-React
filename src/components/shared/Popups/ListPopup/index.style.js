//@flow
import styled from 'styled-components';
import Icon from 'components/shared/Icon';

export const PopupBox = styled.div`
   background-color:${({ backgroundColor }) => backgroundColor}  
   width: ${({ width }) => width};
   height: ${({ height }) => height};
   border-radius: ${({ borderRadius }) => borderRadius};
   margin:0px;
   padding-top:5px;
   padding-bottom: 5px;     
   box-shadow: 0 3px 8px 0 rgba(0, 0, 0, 0.2);
`;

export const IconItem = styled(Icon)`
  margin-right: 8px;
  align-self: center;
`;

export const SubIcon = styled(Icon)`
  margin-left: 46px;
  margin-right: 5px;
  align-self: center;
`;

export const Sublist = styled.div`
  display: none; 
`;

export const ItemWrapper = styled.div`
  display: flex; 
  height: 27px;
  :hover {
    ${Sublist} {
      display: block;
    }
  }
`;
export const PopupItem = styled.div`  
  flex-grow: 1
  flex-shrink: 0;    
  padding: 3px 0px 3px 5px;
  margin: 0px 5px 0px 5px;  
  cursor: pointer;
  :hover {
    background-color: ${({ hoverBackgroundColor }) => hoverBackgroundColor};
    color: ${({ hoverColor }) => hoverColor};
    border-radius: 5px;
    ${IconItem} {
      color: ${({ hoverColorIcon }) => hoverColorIcon};
    }
    ${SubIcon} {
      color: ${({ hoverColorIcon }) => hoverColorIcon};
    }
    }   
  }
`;

export const Line = styled.hr`
  border: 0;
  height: 0;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  border-bottom: 1px solid rgba(255, 255, 255, 0.4);
  margin: 6px 0px 2px 0px;
`;
