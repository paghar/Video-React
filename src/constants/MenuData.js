//@flow

export type MenuDataType = {
  id: number,
  menuIcon: string,
  menuTitle: string,
  menuLink: string,
};

export const MenuData = [
  { id: 1, menuIcon: '61449', menuTitle: 'Workspace', menuLink: '/home' },
  { id: 2, menuIcon: '61474', menuTitle: 'Projects', menuLink: '/home' },
  { id: 3, menuIcon: '61614', menuTitle: 'Board', menuLink: '/home' },
  { id: 4, menuIcon: '61633', menuTitle: 'Links', menuLink: '/home' },
];
