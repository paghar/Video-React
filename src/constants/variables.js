//@flow

export const MEDIA_BROWSER_RIGHT_SIDE_MENU_ANIMATION_DURATION = 300; // in ms
export const LEFT_MENU_ANIMATION_DURATION = 300; // in ms
export const FOLDER_TYPE = 'FOLDER_TYPE';
export const FILE_TYPE = 'FILE_TYPE';
export const POP_UP_LINE = 'HR';
export const POP_UP_HAS_SUBITEMS = 'SUBITEM';
