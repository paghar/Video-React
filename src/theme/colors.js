import theme from 'styled-theming';

export const Generals = {
  i: theme('mode', {
    dark: '#05050d',
  }),
  ii: theme('mode', {
    dark: '#0d0d1a',
  }),
  iii: theme('mode', {
    dark: '#151526',
  }),
  iv: theme('mode', {
    dark: '#212133',
  }),
  v: theme('mode', {
    dark: '#262640',
  }),
  vi: theme('mode', {
    dark: '#7b7b9a',
  }),
  vii: theme('mode', {
    dark: '#b7b7cc',
  }),
  viii: theme('mode', {
    dark: '#ffffff',
  }),
  x: theme('mode', {
    dark: '#e6e6f3',
  }),
  ix: theme('mode', {
    dark: '#f1f1f9',
  }),
};

export const Primary = {
  Dim: theme('mode', {
    dark: '#2172ff',
  }),
  Light: theme('mode', {
    dark: '#317BFC',
  }),
  Dark: theme('mode', {
    dark: '#0E5EE8',
  }),
};

export const PromptWarn = {
  Dim: theme('mode', {
    dark: '#F51653',
  }),
  Light: theme('mode', {
    dark: '#FF4074',
  }),
  Dark: theme('mode', {
    dark: '#DE0540',
  }),
};

export const Secondary = {
  i: theme('mode', {
    dark: '#00d3b2',
  }),
  ii: theme('mode', {
    dark: '#ffcc00',
  }),
  iii: theme('mode', {
    dark: '#f51653',
  }),
  iv: theme('mode', {
    dark: '#64a7ff',
  }),
};

export const Avatars = [
  theme('mode', {
    dark: '#5c4dff',
  }),
  theme('mode', {
    dark: '#685ce6',
  }),
  theme('mode', {
    dark: '#7d73e6',
  }),
  theme('mode', {
    dark: '#928ae6',
  }),
];

export const Gradient = theme('mode', {
  dark:
    'linear-gradient(to bottom, rgba(21, 21, 38, 0), rgba(21, 21, 38, 0.69) 53%, #111120)',
});
