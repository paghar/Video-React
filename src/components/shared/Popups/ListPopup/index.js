//@flow
import React from 'react';
import {
  PopupBox,
  PopupItem,
  IconItem,
  Line,
  SubIcon,
  Sublist,
  ItemWrapper,
} from './index.style';
import { POP_UP_LINE, POP_UP_HAS_SUBITEMS } from 'constants/variables';
import Typography from 'components/shared/Typography';

// itemType  => null:text , hr:line , subItem:have sub items
export type PopupItemType = {
  id: number,
  itemType?: string,
  icon?: string,
  title: string,
  iconColor?: string,
  children: Array<PopupItemType>,
};

type Props = {
  popupItems: Array<PopupItemType>,
  backgroundColor: string,
  width: string,
  height: string,
  borderRadius: string,
  font: Object,
  color: string,
  hoverColor: string,
  hoverBackgroundColor: string,
  iconSize?: string,
  clicked: (item: PopupItemType) => void,
  className?: string,
};
const SimplePopup = ({
  popupItems,
  backgroundColor,
  width,
  height,
  borderRadius,
  font,
  color,
  hoverColor,
  hoverBackgroundColor,
  iconSize,
  clicked,
  className,
}: Props) => {
  return (
    <PopupBox
      backgroundColor={backgroundColor}
      width={width}
      height={height}
      borderRadius={borderRadius}
      className={className}
    >
      {popupItems.map(item => {
        const { id, itemType, icon, title, iconColor, children } = item;
        return itemType == POP_UP_LINE ? (
          <Line key={id}></Line>
        ) : (
          <ItemWrapper key={id}>
            <PopupItem
              onClick={() => clicked(item)}
              hoverColor={hoverColor}
              hoverColorIcon={iconColor || hoverColor}
              hoverBackgroundColor={hoverBackgroundColor}
            >
              {icon && (
                <IconItem
                  fontSize={iconSize}
                  color={iconColor || color}
                  iconNumber={icon}
                ></IconItem>
              )}
              <Typography font={font} color={color}>
                {title}
              </Typography>
              {itemType == POP_UP_HAS_SUBITEMS && (
                <SubIcon
                  fontSize={iconSize}
                  color={color}
                  iconNumber="61524"
                ></SubIcon>
              )}
            </PopupItem>
            {itemType == POP_UP_HAS_SUBITEMS && (
              <Sublist>
                <SimplePopup
                  popupItems={children}
                  backgroundColor={backgroundColor}
                  width="121px"
                  height="199px"
                  borderRadius="7px"
                  font={font}
                  color={color}
                  hoverColor={hoverColor}
                  hoverBackgroundColor={hoverBackgroundColor}
                  clicked={clicked}
                />
              </Sublist>
            )}
          </ItemWrapper>
        );
      })}
    </PopupBox>
  );
};

export default SimplePopup;
