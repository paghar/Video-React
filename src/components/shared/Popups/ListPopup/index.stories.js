import React from 'react';
import { addParameters } from '@storybook/react';
import { themes } from '@storybook/theming';
import Story from '.';
import { Display2 } from 'theme/fonts';
import { action } from '@storybook/addon-actions';
import { POP_UP_LINE, POP_UP_HAS_SUBITEMS } from 'constants/variables';
import { Secondary, Generals } from 'theme/colors';

// itemType  => null:text , hr:line , subItem:have sub items
const SimplePopupData = [
  { id: 1, title: 'Name', children: [] },
  { id: 2, title: 'Size', children: [] },
  { id: 3, title: 'Type', children: [] },
  {
    id: 4,
    title: 'Date uploaded',
    children: [],
  },
  { id: 5, title: 'Status', children: [] },
  { id: 6, title: 'Tag', children: [] },
  { id: 7, title: 'Custom', children: [] },
];

const RightPopupData = [
  {
    id: 1,
    icon: '61920',
    title: 'Share',
    children: [],
  },
  {
    id: 2,
    icon: '61539',
    title: 'Download',
    children: [],
  },
  {
    id: 3,
    icon: '61508',
    title: 'Rename',
    children: [],
  },
  {
    id: 4,
    icon: '62306',
    title: 'Manage versions',
    children: [],
  },
  {
    id: 5,
    icon: '62189',
    title: 'Delete',
    children: [],
  },
];

const StatusPopupData = [
  {
    id: 1,
    icon: '61713',
    title: 'In Progress',
    iconColor: Secondary.iv,
    children: [],
  },
  {
    id: 2,
    icon: '61713',
    title: 'In Review',
    iconColor: Secondary.ii,
    children: [],
  },
  {
    id: 3,
    icon: '61713',
    title: 'Done',
    iconColor: Secondary.i,
    children: [],
  },
  {
    id: 4,
    icon: '61453',
    title: 'No status',
    iconColor: Generals.vi,
    children: [],
  },
];

const SpaceRightPopup = [
  {
    id: 1,
    icon: '61563',
    title: 'New folder',
    children: [],
  },
  {
    id: 2,
    icon: '61459',
    title: 'Project setting',
    children: [],
  },
  {
    id: 3,
    itemType: POP_UP_LINE,
    icon: '61459',
    title: 'Project setting',
    children: [],
  },
  {
    id: 4,
    itemType: POP_UP_HAS_SUBITEMS,
    icon: '62306',
    title: 'Sort by ',
    children: [
      { id: 1, title: 'Name', children: [] },
      { id: 2, title: 'Size', children: [] },
      { id: 3, title: 'Type', children: [] },
      {
        id: 4,
        title: 'Date uploaded',
        children: [],
      },
      { id: 5, title: 'Status', children: [] },
      { id: 6, title: 'Tag', children: [] },
      { id: 7, title: 'Custom', children: [] },
    ],
  },
];

const InviteDropDownList = [
  { id: 1, title: 'Collabrator', children: [] },
  { id: 2, title: 'Project owner', children: [] },
  { id: 3, title: 'Team member', children: [] },
  {
    id: 4,
    title: 'Admin',
    children: [],
  },
];

export default { title: 'Popups' };

addParameters({
  options: {
    theme: themes.light,
  },
});

export const SortPopup = () => (
  <Story
    popupItems={SimplePopupData}
    backgroundColor={Generals.viii}
    width="121px"
    height="199px"
    borderRadius="7px"
    font={Display2}
    color={Generals.vi}
    hoverColor={Generals.ii}
    hoverBackgroundColor={Generals.x}
    clicked={action('clicked')}
  ></Story>
);

export const RightClickPopup = () => (
  <Story
    popupItems={RightPopupData}
    backgroundColor={Generals.viii}
    width="151px"
    height="145x"
    borderRadius="7px"
    font={Display2}
    color={Generals.vi}
    hoverColor={Generals.ii}
    hoverBackgroundColor={Generals.x}
    iconSize="12px"
    clicked={action('clicked')}
  ></Story>
);

export const StatusPopup = () => (
  <Story
    popupItems={StatusPopupData}
    backgroundColor={Generals.viii}
    width="108px"
    height="118x"
    borderRadius="7px"
    font={Display2}
    color={Generals.vi}
    hoverColor={Generals.ii}
    hoverBackgroundColor={Generals.x}
    iconSize="8px"
    clicked={action('clicked')}
  ></Story>
);

export const SpaceRightClickPopup = () => (
  <Story
    popupItems={SpaceRightPopup}
    subPopupItems={SimplePopupData}
    backgroundColor={Generals.viii}
    width="139px"
    height="102px"
    subPopupWidth="121px"
    subPopupHeight="199px"
    borderRadius="7px"
    font={Display2}
    color={Generals.vi}
    hoverColor={Generals.ii}
    hoverBackgroundColor={Generals.x}
    iconSize="12px"
    clicked={action('item')}
  ></Story>
);

export const InvitePeopleDropDownList = () => (
  <Story
    popupItems={InviteDropDownList}
    backgroundColor={Generals.viii}
    width="126px"
    height="118px"
    borderRadius="7px"
    font={Display2}
    color={Generals.vi}
    hoverColor={Generals.ii}
    hoverBackgroundColor={Generals.x}
    clicked={action('clicked')}
  ></Story>
);
