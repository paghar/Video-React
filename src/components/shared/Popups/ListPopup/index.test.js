//@flow
import React from 'react';
import ListPopup from '.';
import renderer from 'react-test-renderer';
import { Display2 } from 'theme/fonts';
import { mount } from 'enzyme';
import { POP_UP_LINE, POP_UP_HAS_SUBITEMS } from 'constants/variables';
import { Secondary, Generals } from 'theme/colors';

// itemType  => null:text , hr:line , subItem:have sub items
const SimplePopupData = [
  { id: 1, title: 'Name', children: [] },
  { id: 2, title: 'Size', children: [] },
  { id: 3, title: 'Type', children: [] },
  {
    id: 4,
    title: 'Date uploaded',
    children: [],
  },
  { id: 5, title: 'Status', children: [] },
  { id: 6, title: 'Tag', children: [] },
  { id: 7, title: 'Custom', children: [] },
];

const RightPopupData = [
  {
    id: 1,
    icon: '61920',
    title: 'Share',
    children: [],
  },
  {
    id: 2,
    icon: '61539',
    title: 'Download',
    children: [],
  },
  {
    id: 3,
    icon: '61508',
    title: 'Rename',
    children: [],
  },
  {
    id: 4,
    icon: '62306',
    title: 'Manage versions',
    children: [],
  },
  {
    id: 5,
    icon: '62189',
    title: 'Delete',
    children: [],
  },
];

const StatusPopupData = [
  {
    id: 1,
    icon: '61713',
    title: 'In Progress',
    iconColor: Secondary.iv,
    children: [],
  },
  {
    id: 2,
    icon: '61713',
    title: 'In Review',
    iconColor: Secondary.ii,
    children: [],
  },
  {
    id: 3,
    icon: '61713',
    title: 'Done',
    iconColor: Secondary.i,
    children: [],
  },
  {
    id: 4,
    icon: '61453',
    title: 'No status',
    iconColor: Generals.vi,
    children: [],
  },
];

const SpaceRightPopup = [
  {
    id: 1,
    icon: '61563',
    title: 'New folder',
    children: [],
  },
  {
    id: 2,
    icon: '61459',
    title: 'Project setting',
    children: [],
  },
  {
    id: 3,
    itemType: POP_UP_LINE,
    icon: '61459',
    title: 'Project setting',
    children: [],
  },
  {
    id: 4,
    itemType: POP_UP_HAS_SUBITEMS,
    icon: '62306',
    title: 'Sort by ',
    children: [
      { id: 1, title: 'Name', children: [] },
      { id: 2, title: 'Size', children: [] },
      { id: 3, title: 'Type', children: [] },
      {
        id: 4,
        title: 'Date uploaded',
        children: [],
      },
      { id: 5, title: 'Status', children: [] },
      { id: 6, title: 'Tag', children: [] },
      { id: 7, title: 'Custom', children: [] },
    ],
  },
];

const InviteDropDownList = [
  { id: 1, title: 'Collabrator', children: [] },
  { id: 2, title: 'Project owner', children: [] },
  { id: 3, title: 'Team member', children: [] },
  {
    id: 4,
    title: 'Admin',
    children: [],
  },
];

describe('<ListPopup />', () => {
  it('click in list item', () => {
    const itemListClick = jest.fn();
    const wrapper = mount(
      <ListPopup
        popupItems={SimplePopupData}
        backgroundColor={Generals.viii}
        width="121px"
        height="199px"
        borderRadius="7px"
        font={Display2}
        color={Generals.vi}
        hoverColor={Generals.ii}
        hoverBackgroundColor={Generals.x}
        clicked={itemListClick}
      ></ListPopup>
    );
    wrapper
      .find('div')
      .at(6)
      .simulate('click');
    expect(itemListClick).toHaveBeenCalled();
    expect(itemListClick.mock.calls[0][0]).toEqual({
      children: [],
      id: 3,
      title: 'Type',
    });
  });

  it('SortPopup', () => {
    const itemListClick = jest.fn();
    const wrapper = renderer
      .create(
        <ListPopup
          popupItems={SimplePopupData}
          backgroundColor={Generals.viii}
          width="121px"
          height="199px"
          borderRadius="7px"
          font={Display2}
          color={Generals.vi}
          hoverColor={Generals.ii}
          hoverBackgroundColor={Generals.x}
          clicked={itemListClick}
        ></ListPopup>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('RightClickPopup', () => {
    const itemListClick = jest.fn();
    const wrapper = renderer
      .create(
        <ListPopup
          popupItems={RightPopupData}
          backgroundColor={Generals.viii}
          width="151px"
          height="145x"
          borderRadius="7px"
          font={Display2}
          color={Generals.vi}
          hoverColor={Generals.ii}
          hoverBackgroundColor={Generals.x}
          iconSize="12px"
          clicked={itemListClick}
        ></ListPopup>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('StatusPopup', () => {
    const itemListClick = jest.fn();
    const wrapper = renderer
      .create(
        <ListPopup
          popupItems={StatusPopupData}
          backgroundColor={Generals.viii}
          width="108px"
          height="118x"
          borderRadius="7px"
          font={Display2}
          color={Generals.vi}
          hoverColor={Generals.ii}
          hoverBackgroundColor={Generals.x}
          iconSize="8px"
          clicked={itemListClick}
        ></ListPopup>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('SpaceRightClickPopup', () => {
    const itemListClick = jest.fn();
    const wrapper = renderer
      .create(
        <ListPopup
          popupItems={SpaceRightPopup}
          subPopupItems={SimplePopupData}
          backgroundColor={Generals.viii}
          width="139px"
          height="102px"
          subPopupWidth="121px"
          subPopupHeight="199px"
          borderRadius="7px"
          font={Display2}
          color={Generals.vi}
          hoverColor={Generals.ii}
          hoverBackgroundColor={Generals.x}
          iconSize="12px"
          clicked={itemListClick}
        ></ListPopup>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('InvitePeopleDropDownList', () => {
    const itemListClick = jest.fn();
    const wrapper = renderer
      .create(
        <ListPopup
          popupItems={InviteDropDownList}
          backgroundColor={Generals.viii}
          width="126px"
          height="118px"
          borderRadius="7px"
          font={Display2}
          color={Generals.vi}
          hoverColor={Generals.ii}
          hoverBackgroundColor={Generals.x}
          clicked={itemListClick}
        ></ListPopup>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
