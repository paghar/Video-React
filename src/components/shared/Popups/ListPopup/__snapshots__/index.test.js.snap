// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListPopup /> InvitePeopleDropDownList 1`] = `
.c0 {
  background-color: width:126px;
  height: 118px;
  border-radius: 7px;
  margin: 0px;
  padding-top: 5px;
  padding-bottom: 5px;
  box-shadow: 0 3px 8px 0 rgba(0,0,0,0.2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 27px;
}

.c1:hover .c4 {
  display: block;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3px 0px 3px 5px;
  margin: 0px 5px 0px 5px;
  cursor: pointer;
}

.c2:hover {
  border-radius: 5px;
}

.c3 {
  pointer-events: auto;
}

<div
  className="c0"
  height="118px"
  width="126px"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Collabrator
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Project owner
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Team member
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Admin
      </span>
    </div>
  </div>
</div>
`;

exports[`<ListPopup /> RightClickPopup 1`] = `
.c3 {
  font-family: 'FontAwesome' !important;
  font-size: 12px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  background-color: width:151px;
  height: 145x;
  border-radius: 7px;
  margin: 0px;
  padding-top: 5px;
  padding-bottom: 5px;
  box-shadow: 0 3px 8px 0 rgba(0,0,0,0.2);
}

.c4 {
  margin-right: 8px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 27px;
}

.c1:hover .c6 {
  display: block;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3px 0px 3px 5px;
  margin: 0px 5px 0px 5px;
  cursor: pointer;
}

.c2:hover {
  border-radius: 5px;
}

.c5 {
  pointer-events: auto;
}

<div
  className="c0"
  height="145x"
  width="151px"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        Share
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        Download
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        Rename
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        Manage versions
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        Delete
      </span>
    </div>
  </div>
</div>
`;

exports[`<ListPopup /> SortPopup 1`] = `
.c0 {
  background-color: width:121px;
  height: 199px;
  border-radius: 7px;
  margin: 0px;
  padding-top: 5px;
  padding-bottom: 5px;
  box-shadow: 0 3px 8px 0 rgba(0,0,0,0.2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 27px;
}

.c1:hover .c4 {
  display: block;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3px 0px 3px 5px;
  margin: 0px 5px 0px 5px;
  cursor: pointer;
}

.c2:hover {
  border-radius: 5px;
}

.c3 {
  pointer-events: auto;
}

<div
  className="c0"
  height="199px"
  width="121px"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Name
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Size
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Type
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Date uploaded
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Status
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Tag
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Custom
      </span>
    </div>
  </div>
</div>
`;

exports[`<ListPopup /> SpaceRightClickPopup 1`] = `
.c3 {
  font-family: 'FontAwesome' !important;
  font-size: 12px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  background-color: width:121px;
  height: 199px;
  border-radius: 7px;
  margin: 0px;
  padding-top: 5px;
  padding-bottom: 5px;
  box-shadow: 0 3px 8px 0 rgba(0,0,0,0.2);
}

.c0 {
  background-color: width:139px;
  height: 102px;
  border-radius: 7px;
  margin: 0px;
  padding-top: 5px;
  padding-bottom: 5px;
  box-shadow: 0 3px 8px 0 rgba(0,0,0,0.2);
}

.c4 {
  margin-right: 8px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  margin-left: 46px;
  margin-right: 5px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c9 {
  display: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 27px;
}

.c1:hover .c8 {
  display: block;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3px 0px 3px 5px;
  margin: 0px 5px 0px 5px;
  cursor: pointer;
}

.c2:hover {
  border-radius: 5px;
}

.c6 {
  border: 0;
  height: 0;
  border-top: 1px solid rgba(0,0,0,0.1);
  border-bottom: 1px solid rgba(255,255,255,0.4);
  margin: 6px 0px 2px 0px;
}

.c5 {
  pointer-events: auto;
}

<div
  className="c0"
  height="102px"
  width="139px"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        New folder
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        Project setting
      </span>
    </div>
  </div>
  <hr
    className="c6"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="12px"
      >
        
      </span>
      <span
        className="c5"
      >
        Sort by 
      </span>
      <span
        className="c3 c7"
        fontSize="12px"
      >
        
      </span>
    </div>
    <div
      className="c8 c9"
    >
      <div
        className="c10"
        height="199px"
        width="121px"
      >
        <div
          className="c1"
        >
          <div
            className="c2"
            onClick={[Function]}
          >
            <span
              className="c5"
            >
              Name
            </span>
          </div>
        </div>
        <div
          className="c1"
        >
          <div
            className="c2"
            onClick={[Function]}
          >
            <span
              className="c5"
            >
              Size
            </span>
          </div>
        </div>
        <div
          className="c1"
        >
          <div
            className="c2"
            onClick={[Function]}
          >
            <span
              className="c5"
            >
              Type
            </span>
          </div>
        </div>
        <div
          className="c1"
        >
          <div
            className="c2"
            onClick={[Function]}
          >
            <span
              className="c5"
            >
              Date uploaded
            </span>
          </div>
        </div>
        <div
          className="c1"
        >
          <div
            className="c2"
            onClick={[Function]}
          >
            <span
              className="c5"
            >
              Status
            </span>
          </div>
        </div>
        <div
          className="c1"
        >
          <div
            className="c2"
            onClick={[Function]}
          >
            <span
              className="c5"
            >
              Tag
            </span>
          </div>
        </div>
        <div
          className="c1"
        >
          <div
            className="c2"
            onClick={[Function]}
          >
            <span
              className="c5"
            >
              Custom
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ListPopup /> StatusPopup 1`] = `
.c3 {
  font-family: 'FontAwesome' !important;
  font-size: 8px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  background-color: width:108px;
  height: 118x;
  border-radius: 7px;
  margin: 0px;
  padding-top: 5px;
  padding-bottom: 5px;
  box-shadow: 0 3px 8px 0 rgba(0,0,0,0.2);
}

.c4 {
  margin-right: 8px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 27px;
}

.c1:hover .c6 {
  display: block;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3px 0px 3px 5px;
  margin: 0px 5px 0px 5px;
  cursor: pointer;
}

.c2:hover {
  border-radius: 5px;
}

.c5 {
  pointer-events: auto;
}

<div
  className="c0"
  height="118x"
  width="108px"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="8px"
      >
        
      </span>
      <span
        className="c5"
      >
        In Progress
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="8px"
      >
        
      </span>
      <span
        className="c5"
      >
        In Review
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="8px"
      >
        
      </span>
      <span
        className="c5"
      >
        Done
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3 c4"
        fontSize="8px"
      >
        
      </span>
      <span
        className="c5"
      >
        No status
      </span>
    </div>
  </div>
</div>
`;
